src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -23,7 +23,7 @@ public sealed class CompositeActionHandler : Handler, ICompositeActionHandler
     {
         public CompositeActionExecutionData Data { get; set; }
 
-        public Task RunAsync(ActionRunStage stage)
+        public async Task RunAsync(ActionRunStage stage)
```

Comment:
Can the "RegisterNestedSteps" method be deleted now from ExecutionContext?

And can we delete the CompositeActionOutputHandler too?

Reply:
We do `Root.CreateChild(_record.Id, step.DisplayName,...` in the RegisterNestedStep. `Root` is private to the ExecutionContext so we can't just move that logic into the handler.